lib/zip/file.rb

```diff
@@ -229,12 +231,14 @@ def save_splited_part(zip_file, partial_zip_file_name, zip_file_size, szip_file_
       def split(zip_file_name, segment_size = MAX_SEGMENT_SIZE, delete_zip_file = true, partial_zip_file_name = nil)
         raise Error, "File #{zip_file_name} not found" unless ::File.exist?(zip_file_name)
         raise Errno::ENOENT, zip_file_name unless ::File.readable?(zip_file_name)
+
         zip_file_size = ::File.size(zip_file_name)
         segment_size  = get_segment_size_for_split(segment_size)
         return if zip_file_size <= segment_size
+
         segment_count = get_segment_count_for_split(zip_file_size, segment_size)
         # Checking for correct zip structure
-        open(zip_file_name) {}
+        ::Zip::File.open(zip_file_name) {}
```

Comment:
Hmm. Rubocop certainly complains about it if it's left as just `open`. `self.open` might be a better bet?

Reply:
Looking again at the whole file, using `::Zip::File` to disambiguate is common practice, so I think best to leave this for consistency. Rubocop considers bare `open`s to be a security flaw.

Reply:
Sorry, I misunderstood this, and the comment was vague. I thought the line was redundant (since it just opens the file and then closes it again), but as the comment above explains, it's checking that it's valid.

FWIW it was `self.open` before https://github.com/rubyzip/rubyzip/commit/abf9cbec23e1a0842d755bd1505f564547e32674, another rubocop fix, but I think this is fine.

Reply:
Yes, I had to double-check that it was actually doing something - side-effects, eh?